### PR TITLE
Schematics: Use MockStore in container generated tests

### LINKED ISSUE #2028: Schematics: Use MockStore in container generated tests
The test file generated by the container schematic is currently importing the StoreModule:
```ts
// ...
import { Store, StoreModule } from '@ngrx/store';

describe('ExampleComponent', () => {
  // ...
  let store: Store<any>;

  beforeEach(async() => {
    TestBed.configureTestingModule({
      imports: [ StoreModule.forRoot({}) ], // <--- Outdated
      declarations: [ ExampleComponent ]
    });

    await TestBed.compileComponents();
  });

  beforeEach(() => {
    // ...
    store = TestBed.get<Store>(Store);

    spyOn(store, 'dispatch').and.callThrough();
    fixture.detectChanges();
  });
  // ...
});
```

Instead, the schematic should use MockStore:
```ts
// ...
import { provideMockStore, MockStore } from '@ngrx/store/testing';

describe('ExampleComponent', () => {
  // ...
  let store: MockStore<any>;

  beforeEach(async() => {
    TestBed.configureTestingModule({
      providers: [ provideMockStore() ],
      declarations: [ ExampleComponent ]
    });

    await TestBed.compileComponents();
  });

  beforeEach(() => {
    // ...
    store = TestBed.get(Store);
    fixture.detectChanges();
  });
  // ...
});
```

## If accepted, I would be willing to submit a PR for this feature

[x] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by jtcrowson**
## PR Checklist 

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

- Container schematic uses StoreModule for unit tests

Closes #2028

## What is the new behavior?

- Container schematic uses MockStore for unit tests
- `testDepth` option to use StoreModule for integration tests instead.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/schematics/src/container/files/__name@dasherize@if-flat__/__name@dasherize__.component.spec.ts.template [hunk 1]
```diff
@@ -20,9 +20,8 @@ describe('<%= classify(name) %>Component', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(<%= classify(name) %>Component);
     component = fixture.componentInstance;
-    store = TestBed.get<Store>(Store);
+    store = TestBed.get(Store);
```
Comment (timdeschryver) [inline modules/schematics/src/container/files/__name@dasherize@if-flat__/__name@dasherize__.component.spec.ts.template]: We should use the generic `TestBed.get`, as this is [deprecated](https://angular.io/api/core/testing/TestBed#get)

Comment (jtcrowson) [reply modules/schematics/src/container/files/__name@dasherize@if-flat__/__name@dasherize__.component.spec.ts.template]: ah ok, I was wondering why that changed.  Fixed in 689fc93.

Doing a quick file search, looks like we have a bunch of code across the project that still uses the non-generic form.  Want me to create an issue/open a PR for it?

**Comment by brandonroberts** [comment]
What do you think about providing an option to generate an integration test vs a unit test instead of replacing it?

**Comment by jtcrowson** [comment]
I’m up for it. Should we use two separate options?
`--skipTests=true|false`, default: false (existing in angular/schematics)
`--testDepth=unit|integration`, default: unit (new for NgRx/schematics)

**Comment by brandonroberts** [comment]
👍

File: modules/schematics/src/container/files/__name@dasherize@if-flat__/__name@dasherize__.component.spec.ts.template (hunk 1 above)
Comment (timdeschryver) [reply modules/schematics/src/container/files/__name@dasherize@if-flat__/__name@dasherize__.component.spec.ts.template]: That's fine by me @jtcrowson 👍

**Comment by jtcrowson** [comment]
@brandonroberts Added the `testDepth` option in 896899c and b696b66  👍

File: modules/schematics/src/container/schema.json [hunk 2]
```diff
@@ -97,6 +97,13 @@
       "type": "string",
       "default": "State",
       "description": "Specifies the interface for the state."
+    },
+    "testDepth": {
+      "description":
+        "Specifies whether to create a unit test or an integration test.",
+      "enum": ["Unit", "Integration"],
```
Comment (brandonroberts) [inline modules/schematics/src/container/schema.json]:
```suggestion
      "enum": ["unit", "integration"],
```

File: modules/schematics/src/container/schema.json [hunk 3]
```diff
@@ -97,6 +97,13 @@
...
+      "type": "string",
+      "default": "Unit"
```
Comment (brandonroberts) [inline modules/schematics/src/container/schema.json]: The current setup should remain the default for now.

```suggestion
      "default": "integration"
```

File: projects/ngrx.io/content/guide/schematics/container.md [hunk 4]
```diff
@@ -35,6 +35,13 @@ Provide the name of the interface exported for your state interface
   - Type: `string`
   - Default: `State`
 
+Specifies whether to create a unit test or an integration test
+
+- `--testDepth`
+  - Type: `string`
+  - Values: `Unit|Integration`
+  - Default: `Unit`
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/schematics/container.md]:
```suggestion
  - Default: `integration`
```

File: projects/ngrx.io/content/guide/schematics/container.md [hunk 5]
```diff
@@ -35,6 +35,13 @@ Provide the name of the interface exported for your state interface
   - Type: `string`
   - Default: `State`
 
+Specifies whether to create a unit test or an integration test
+
+- `--testDepth`
+  - Type: `string`
+  - Values: `Unit|Integration`
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/schematics/container.md]:
```suggestion
  - Values: `unit|integration`
```

File: modules/schematics/src/container/index.ts [hunk 6]
```diff
@@ -136,17 +136,22 @@ export default function(options: ContainerOptions): Rule {
...
-      applyTemplates({
-        'if-flat': (s: string) => (options.flat ? '' : s),
-        ...stringUtils,
-        ...(options as object),
-      } as any),
-      move(parsedPath.path),
-    ]);
+    const templateSource = apply(
+      url(
+        options.testDepth === 'Integration' ? './integration-files' : './files'
```
Comment (brandonroberts) [inline modules/schematics/src/container/index.ts]:
```suggestion
        options.testDepth === 'integration' ? './integration-files' : './files'
```

**Comment by jtcrowson** [comment]
@brandonroberts just curious, why lowercase for the enums?  Looks like angular uses uppercase for enums that are words: https://angular.io/cli/generate

**Comment by brandonroberts** [comment]
For the values? The key is `testDepth` in camel case, which matches the CLI, but the values are lowercase.

**Comment by brandonroberts** [comment]
I see it now. I think that one is out of place. I'd rather keep the consistent with lowercase for the values.

**Comment by brandonroberts** [comment]
@jtcrowson will you check the failing unit tests?

**Comment by jtcrowson** [comment]
@brandonroberts sorry for the delay, I've been busy with Angular Denver stuff.

I update the integration/unit casing and made integration the default.

I'm having trouble debugging the test failures since I can't get `yarn test:bazel` running locally.  The CircleCI print out isn't helping much either.

**Comment by brandonroberts** [comment]
No problem. I'll take a look at the test failures.

**Comment by jtcrowson** [comment]
@brandonroberts thanks for d713cbe, makes sense why it was passing with jasmine locally but not with bazel.

**Comment by brandonroberts** [comment]
Yep, its weird that it just fails silently also, and doesn't generate the files if it doesn't find the path.

**Pull request merged**